packages/itwinui-react/src/core/ExpandableBlock/ExpandableBlock.tsx

```diff
@@ -120,43 +154,178 @@ export const ExpandableBlock = React.forwardRef((props, ref) => {
       {...rest}
     >
+      {children ?? <SvgChevronRight aria-hidden />}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+ExpandableBlockExpandIcon.displayName = 'ExpandableBlock.ExpandIcon';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.LabelArea component
+
+const ExpandableBlockLabelArea = polymorphic.span('iui-expandable-block-label');
+ExpandableBlockLabelArea.displayName = 'ExpandableBlock.LabelArea';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Title component
+
+const ExpandableBlockTitle = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  return (
+    <Box
+      className={cx('iui-expandable-block-title', className)}
+      ref={forwardedRef}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div'>;
+ExpandableBlockTitle.displayName = 'ExpandableBlock.Title';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Caption component
+
+const ExpandableBlockCaption = polymorphic('iui-expandable-block-caption');
+ExpandableBlockCaption.displayName = 'ExpandableBlock.Caption';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.EndIcon component
+
+const ExpandableBlockEndIcon = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  const { status } = useSafeContext(ExpandableBlockContext);
+
+  const icon = children ?? (status && StatusIconMap[status]());
+  return (
+    <Icon className={className} fill={status} ref={forwardedRef} {...rest}>
+      {icon}
+    </Icon>
+  );
+}) as PolymorphicForwardRefComponent<
+  'span',
+  React.ComponentPropsWithoutRef<typeof Icon>
+>;
+ExpandableBlockEndIcon.displayName = 'ExpandableBlock.EndIcon';
+
+// ----------------------------------------------------------------------------
+// ExpandableBlock.Content component
+
+const ExpandableBlockContent = React.forwardRef((props, forwardedRef) => {
+  const { className, children, ...rest } = props;
+  const { isExpanded } = useSafeContext(ExpandableBlockContext);
+
+  return (
+    <WithCSSTransition in={isExpanded}>
       <Box
-        role='button'
-        aria-expanded={expanded}
-        className='iui-header'
-        aria-disabled={disabled}
-        tabIndex={0}
-        onClick={handleToggle}
-        onKeyDown={onKeyDown}
+        className={cx('iui-expandable-content', className)}
+        ref={forwardedRef}
+        {...rest}
       >
-        <SvgChevronRight className='iui-icon' aria-hidden />
-        <Box as='span' className='iui-expandable-block-label'>
-          <Box className='iui-title'>{title}</Box>
-          {caption && <Box className='iui-caption'>{caption}</Box>}
-        </Box>
-        {icon && <Icon fill={status}>{icon}</Icon>}
+        <Box>{children}</Box>
```

Comment:
lets add `innerProps` if user wants to add a class to this div

Reply:
where is `innerProps` passed in from? Do you mean like this? 
```
const ExpandableBlockContent = React.forwardRef((props, forwardedRef) => {
  const { className, children, innerProps, ...rest } = props;
  ```

Reply:
yes, and used here:

```jsx
<Box {...innerProps}>{children}</Box>
```

passed by user like this:

```jsx
<ExpandableBlock.Content innerProps={{ className: 'my-inner' }}>
```

Reply:
Ok I added innerProps

Reply:
This will be better:
```
innerProps: React.ComponentPropsWithoutRef<'div'>;
```

Reply:
thanks! updated